src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -85,34 +85,41 @@ assign(ReactDOMTextComponent.prototype, {
       if (parentInfo) {
         // parentInfo should always be present except for the top-level
         // component when server rendering
-        validateDOMNesting('span', this, parentInfo);
+        validateDOMNesting('#text', this, parentInfo);
       }
     }
 
     var domID = nativeContainerInfo._idCounter++;
+    var openingValue = ' react-text: ' + domID + ' ';
+    var closingValue = ' /react-text ';
     this._domID = domID;
     this._nativeParent = nativeParent;
     if (transaction.useCreateElement) {
       var ownerDocument = nativeContainerInfo._ownerDocument;
-      var el = ownerDocument.createElement('span');
-      ReactDOMComponentTree.precacheNode(this, el);
-      var lazyTree = DOMLazyTree(el);
-      DOMLazyTree.queueText(lazyTree, this._stringText);
-      return lazyTree;
+      var openingComment = ownerDocument.createComment(openingValue);
+      var closingComment = ownerDocument.createComment(closingValue);
+      this._closingComment = closingComment;
+      var fragment = ownerDocument.createDocumentFragment();
```

Comment:
Though it might be nice to avoid appending in the case that DOMLazyTree is in its lazy mode (in IE) since then we'll be reparenting these nodes twice (see the comments in that file explaining if you haven't already). It's not a huge issue because they don't have children but fewer DOM operations is still usually better.